README.md

```diff
@@ -184,6 +186,58 @@ This module includes a lifecycle management feature that ensures graceful termin
 
 To enable lifecycle management, provide the `spacelift_api_credentials` variable and configure the `instance_refresh` variable. See the [BYO SSM and Secrets Manager with Autoscaling and Lifecycle example](./examples/byo-ssm-secretsmanager-with-autoscaling-and-lifecycle/) for a complete configuration.
 
+## 💰 Spot Instances for Cost Optimization
+
+This module supports AWS Spot Instances, which can provide significant cost savings (up to 90%) compared to On-Demand instances. Spot instances use spare AWS capacity and are ideal for fault-tolerant and flexible workloads.
+
+> ⚠️ Spot instances are **NOT recommended for critical workloads** as they can be interrupted with only 2 minutes notice, potentially causing:
+> - Incomplete or corrupted Terraform state
+> - Failed deployments leaving infrastructure in inconsistent state
+> - Loss of work-in-progress for long-running operations
+
+### How to Enable Spot Instances
+
+Configure spot instances using the `instance_market_options` variable:
+
+```hcl
+module "spacelift_workerpool" {
+  source = "github.com/spacelift-io/terraform-aws-spacelift-workerpool-on-ec2"
+
+  # ... other configuration ...
+
+  # Enable spot instances
+  instance_market_options = {
+    market_type = "spot"
+    # spot_options = {
+    #  spot_instance_type             = "one-time"
+    #  instance_interruption_behavior = "terminate"
+    # }
+  }
+}
+```
+
+### Configuration Options
+
+- **max_price** (Optional): Maximum hourly price you're willing to pay. AWS recommends omitting this to use current Spot pricing, as setting a lower price can increase interruption frequency.
+- **spot_instance_type**: Use `"one-time"` for Auto Scaling Groups (recommended) or `"persistent"` for individual instances.
+- **instance_interruption_behavior**: How instances behave when interrupted - `"terminate"` (default), `"stop"`, or `"hibernate"`. For AutoScaling Groups, it's recommended to use `"terminate"`, as the ASG handles replacements automatically.
+
+These options use sensible defaults when omitted, so explicit configuration is typically unnecessary.
+
+### Graceful Spot Instance Interruption Handling
+
+The Spacelift worker includes built-in spot instance interruption detection to minimize job disruption:
```

Comment:
This is a recent development in the worker code that I did last night. I'm still testing it.

Reply:
just tested it and it works :)